server/src/controllers/course.ts

```diff
@@ -20,6 +23,16 @@ export interface CourseData {
   evaluationInformation: LocalizedString
 }
 
+export interface InstanceData extends CourseData {
```

Comment:
Is the purpose of this interface to match what Sisu returns, or is it rather meant for our internal representation purposes? If it's the latter, I would prefer to use composition over inheritance here.

Reply:
you mean using classes instead of interface?

Reply:
I think he means having the `CourseData` be a member of `InstanceData` rather than `InstanceData` extending it:
```
interface InstanceData {
  courseData: CourseData,
  ...
}
```

Reply:
> I think he means having the `CourseData` be a member of `InstanceData` rather than `InstanceData` extending it:
> 
> ```
> interface InstanceData {
>   courseData: CourseData,
>   ...
> }
> ```

Follow up question. What if we would use class here?
Something like:
```JavaScript
// class for course
class CourseData {
  id: number;
  courseCode: string;
  minCredits: number;
  maxCredits: number;
  department: LocalizedString;
  name: LocalizedString;
  evaluationInformation: LocalizedString;

  constructor(course: Course) {
    this.id = course.id;
    this.courseCode = course.courseCode;
    this.minCredits = course.minCredits;
    this.maxCredits = course.maxCredits;
    // + Other data parser for translations, instances etc.
  }
}

// route function
const course: Course = await courseService.findCourseById(courseId);
const parsedCourse: CourseData = new CourseData(course);
````

Different branches implementing different routes are now using variety of ways to parse the translation and course instance data. At some point we should try to refactor the code for reusability, could a class approach solve this? Or does this make any sense?

Reply:
updated https://github.com/aalto-grades/base-repository/pull/112/commits/19a2ea7e85569ff5151f0978bfcf930351875a72

Reply:
Awesome, this is what I was going for!